secret-sharing/src/churp/player.rs

```diff
@@ -53,20 +61,62 @@ where
         p: &Polynomial<D::PrimeField>,
         vm: &VerificationMatrix<D::Group>,
     ) -> Result<Player<D>> {
-        if p.degree() != self.p.degree() {
+        if p.degree() != self.share.p.degree() {
             return Err(Error::PolynomialDegreeMismatch.into());
         }
         if !vm.is_zero_hole() {
             return Err(Error::VerificationMatrixZeroHoleMismatch.into());
         }
-        if vm.dimensions() != self.vm.dimensions() {
+        if vm.dimensions() != self.share.vm.dimensions() {
             return Err(Error::VerificationMatrixDimensionMismatch.into());
         }
 
-        let p = p + &self.p;
-        let vm = vm + &self.vm;
+        let p = p + &self.share.p;
+        let vm = vm + &self.share.vm;
         let player = Player::new(p, vm);
 
         Ok(player)
     }
 }
+
+impl<D> From<SecretShare<D::Group>> for Player<D>
+where
+    D: DealerParams,
+{
+    fn from(state: SecretShare<D::Group>) -> Player<D> {
+        Player { share: state }
+    }
+}
+
+/// Secret share of the shared secret.
+#[derive(PartialEq, Eq)]
```

Comment:
Where do you need to do comparisons? These will not be constant-time so can leak secret state.

Reply:
In tests only. Agree, need to remove this.

Reply:
You can do it only for test builds:
```rust
#[cfg_attr(test, derive(PartialEq, Eq))]
```

Reply:
Tnx, didn't know that. This is needed in the next part, so will remove for now.